tests/Unit/Provider/MailProviderTest.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Tests\Unit\Provider;
+
+use ChristophWurst\Nextcloud\Testing\TestCase;
+use OCA\Mail\Account;
+use OCA\Mail\Db\MailAccount;
+use OCA\Mail\Exception\ClientException;
+use OCA\Mail\Provider\MailProvider;
+use OCA\Mail\Provider\MailService;
+use OCA\Mail\Service\AccountService;
+use OCP\IL10N;
+use OCP\Mail\Provider\Address as MailAddress;
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+class MailProviderTest extends TestCase {
+
+	/** @var ContainerInterface&MockObject */
+	private $containerInterface;
+	/** @var AccountService&MockObject */
+	private $accountService;
+	/** @var LoggerInterface */
+	private $logger;
+	/** @var IL10N&MockObject */
+	private $l10n;
```

Comment:
```suggestion
	private ContainerInterface&MockObject $containerInterface;
	private AccountService&MockObject $accountService;
	private LoggerInterface $logger;
	private IL10N&MockObject $l10n;
```

Reply:
Mistro you asked me to comment these variables this way and now you want them a different way?

Reply:
Sorry, it wasnt my intention to sent you on a wrong track. That wasn't in this pr, right?

Rule of thumb: Native type declarations over phpdoc annotations.